virtual_rainforest/models/soil/soil_model.py

```diff
@@ -132,29 +147,122 @@ def spinup(self) -> None:
-        dt = self.update_interval.astype("timedelta64[s]").astype(float) / (
-            60.0 * 60.0 * 24.0
-        )
-
-        carbon_pool_updates = self.carbon.calculate_soil_carbon_updates(
-            self.data,
-            self.data["pH"],
-            self.data["bulk_density"],
-            self.data["soil_moisture"],
-            self.data["soil_temperature"],
-            self.data["percent_clay"],
-            dt,
-        )
+        # Find carbon pool updates by integration
+        updated_carbon_pools = self.integrate_soil_model()
 
         # Update carbon pools (attributes and data object)
         # n.b. this also updates the data object automatically
-        self.carbon.update_soil_carbon_pools(self.data, carbon_pool_updates)
+        self.replace_soil_pools(updated_carbon_pools)
 
         # Finally increment timing
         self.next_update += self.update_interval
 
     def cleanup(self) -> None:
         """Placeholder function for soil model cleanup."""
+
+    def replace_soil_pools(self, new_pools: Dataset) -> None:
+        """Replace soil pools with previously calculated new pool values.
+
+        The state of particular soil pools will effect the rate of other processes in
+        the soil module. These processes in turn can effect the exchange rates between
+        the original soil pools. Thus, a separate update function is necessary so that
+        the new values for all soil module components can be calculated on a single
+        state, which is then updated (using this function) when all values have been
+        calculated.
+
+        Args:
+            new_pools: Array of new pool values to insert into the data object
+        """
+
+        self.data["low_molecular_weight_c"] = new_pools["new_lmwc"]
+        self.data["mineral_associated_om"] = new_pools["new_maom"]
+
+    def integrate_soil_model(self) -> Dataset:
+        """Function to integrate the soil model.
+
+        For now a single integration will be used to advance the entire soil module.
+        However, this might get split into several separate integrations in future (if
+        that is feasible).
+
+        This function unpacks the variables that are to be integrated into a single
+        numpy array suitable for integration. This order of this numpy array is as
+        follows: [low_molecular_weight_c, mineral_associated_om].
+
+        Returns:
+            A data array containing the new pool values (i.e. the values at the final
+            time point)
+
+        Raises:
+            IntegrationError: When the integration cannot be successfully completed.
+        """
+
+        # Find number of grid cells integration is being performed over
+        no_cells = self.data.grid.n_cells
+
+        # Extract update interval
+        update_time = self.update_interval.astype("timedelta64[s]").astype(float)
+        # Convert from second to day units
+        t_span = (0.0, update_time / (60.0 * 60.0 * 24.0))
```

Comment:
I *think* you can do this:

```suggestion
        t_span = (0.0, np.timedelta64(update_time, "D"))
```

Reply:
Not that this gives the floor of the number of days. Which might be appropriate, but equally it might be an error for this not to be an exact number of days. Although leap seconds...

Reply:
The docs for `solve_ivp` won't accept `timedelta64` as an input.

I can get rid of that unpleasant hard coded sum by just calculating `update_time` as
```python
update_time = self.update_interval / np.timedelta64(1, "D")
```
which returns a float

Reply:
I've run into the same issue with `scipy.interpolate.interp1d`: the interpolation axis inputs need converting to floatlike things first. That division trick is very clean, although I still think timespans that aren't multiples of a day are probably trouble 😄

Reply:
I don't think time steps that aren't multiple of days are a problem, e.g.

```python
from numpy import timedelta64

timedelta64(5, "h") / timedelta64(1, "D")
```

results in 0.20833333333333334.

Leap seconds are still a potential problem, but I presume that's a were are a few seconds wrong problem rather than something that causes `timedelta64` to catch fire and die?

Reply:
Oh sure,  the division works fine, but I'm just unsure whether models should be using, say,  an update spacing of 7.6734 days. 😄 I don't know if we want to trap that kind of thing.

Reply:
I'm guessing if we want to trap it we would want to trap it higher in the simulation flow, trapping it here would involve trapping it for every model individually